sdk/core/Microsoft.Azure.Core.NewtonsoftJson/src/Serialization/NewtonsoftJsonObjectSerializer.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// A <see cref="NewtonsoftJsonObjectSerializer"/> implementation that uses <see cref="JsonSerializer"/> to for serialization/deserialization.
+    /// </summary>
+    public class NewtonsoftJsonObjectSerializer : ObjectSerializer, ISerializedMemberNameProvider
+    {
+        private const int DefaultBufferSize = 1024;
+
+        // Older StreamReader and StreamWriter would otherwise default to this.
+        private static readonly Encoding UTF8NoBOM = new UTF8Encoding(false, true);
+
+        private readonly JsonSerializer _serializer;
+
+        /// <summary>
+        /// Initializes new instance of <see cref="NewtonsoftJsonObjectSerializer"/> using <see cref="JsonConvert.DefaultSettings"/>.
+        /// </summary>
+        public NewtonsoftJsonObjectSerializer() : this(true)
+        {
+        }
+
+        /// <summary>
+        /// Initializes new instance of <see cref="NewtonsoftJsonObjectSerializer"/>.
+        /// </summary>
+        /// <param name="useDefaultSettings">Whether to use <see cref="JsonConvert.DefaultSettings"/> unless overridden with any given <paramref name="settingsOverrides"/>.</param>
+        /// <param name="settingsOverrides">The <see cref="JsonSerializerSettings"/> instance to use when serializing/deserializing.</param>
+        public NewtonsoftJsonObjectSerializer(bool useDefaultSettings, JsonSerializerSettings? settingsOverrides = null)
+        {
+            _serializer = useDefaultSettings ?
+                JsonSerializer.CreateDefault(settingsOverrides) :
+                JsonSerializer.Create(settingsOverrides);
+        }
```

Comment:
This `bool` feels a little weird.  Is it common to use?  Maybe this would be better as three .ctors like:
```C#
public NewtonsoftJsonObjectSerializer() { }
public NewtonsoftJsonObjectSerializer(JsonSerializerSettings settings) { }
public NewtonsoftJsonObjectSerializer(JsonSerializerSettings settings, bool overrideDefaults) { }
```

Reply:
Json.NET has two ways of creating a serializer. If you look at https://github.com/Azure/azure-sdk-for-net/pull/13527/files/271e2558f1a0d5bbd9b7562df4323cd7615f112b#diff-4f300457184c6cc1fcf3f287fb810e58R41-R43, it can merge whatever settings you provide with defaults, or not and use only what you pass (though, ultimately, it still defaults much of the code where it needs to, like for `IContractResolver`).

So the `settings` here are actually the overrides to optional defaults.

The method to "merge the settings is internal, nor is it common to need it anyway. The other option would be to have them pass in the `JsonSerializer` itself.

Reply:
We could also just always use `CreateDefault` after digging into this more. It uses `JsonConvert.DefaultSettings`, which you can use as a global settings. But if it's null - and it is null by default - it has the same effect as calling `Create`. Devs could provide all the same defaults to `settings` to override anything they needed to override either way. But if we do that, we should at least document which method we're calling and what they can do about it.

Reply:
Of course, another way of looking at it is: because `CreateDefault` uses global `JsonConvert.DefaultSettings`, we shouldn't trust it and only call `Create` with any settings they pass.

Reply:
Another option is just to take the `JsonSerializer` and let the dev create it, but that seems like a strange departure from creating the STJ serializer for its implementation (granted: that's easier to create).

Reply:
Talked to James Newton-King, who suggested - since we allow passing a `JsonSerializerSettings` - to just call `Create`.